rbx_types/src/unique_id.rs

```diff
@@ -68,14 +81,38 @@ impl UniqueId {
         })
     }
 
+    /// Returns whether this `UniqueId` represents a value that **must** be
+    /// unique across all Instances in a Roblox file.
+    pub fn is_unique(&self) -> bool {
+        // Note that because only one field needs to be non-zero, we use
+        // || here
+        self.time != 0 || self.index != 0 || self.random != 0
+    }
+
+    /// Returns whether this `UniqueId` represents the non-unique
+    /// `UniqueId` value.
+    pub fn is_non_unique(&self) -> bool {
+        !self.is_unique()
+    }
+
```

Comment:
Again, I think `is_nil` would be a better better fit here, since `is_unique` doesn't actually do any validation of uniqueness (in other words, non-unique UniqueIds are not necessarily all zeros) and could be misleading

And two methods are probably unnecessary, I think just having `is_nil` and making consumers do `!unique_id.is_nil()` for the inverse is fine